src/wallet/rpcwallet.cpp

```diff
@@ -2034,10 +2034,13 @@ UniValue getwalletinfo(const UniValue& params, bool fHelp)
     obj.pushKV("paytxfee",      ValueFromAmount(payTxFee.GetFeePerK()));
     auto mnemonicChain = pwalletMain->GetMnemonicHDChain();
     if (mnemonicChain.has_value())
-         obj.pushKV("seedfp", mnemonicChain.value().GetSeedFingerprint().GetHex());
-    auto legacyChain = pwalletMain->GetLegacyHDChain();
-    if (legacyChain.has_value())
-         obj.pushKV("legacy_seedfp", legacyChain.value().GetSeedFingerprint().GetHex());
+         obj.pushKV("mnemonic_seedfp", mnemonicChain.value().GetSeedFingerprint().GetHex());
+    // TODO: do we really need to return the legacy seed fingerprint if we're
+    // no longer using it to generate any new keys? What do people actually use
+    // the fingerprint for?
```

Comment:
I don't know what other people use it for, but to me, if I see a given seed fp in two wallets (either directly or because I've written one of them down whether or not I still have that wallet), then I can infer that those wallets have generated and will generate the same keys. For that I do need the legacy seed fp if it has ever been used.